### PR TITLE
Add autoBasePath

### PR DESCRIPTION
**Pull request by abdonrd**
This new flag sets the entrypoint `<base>` tag for all builds to match the name of that build. Unlike the other flags, it does not necessarily trigger a single one-off build. Instead, it switches the option on for all builds (including a one-off, if another flag triggered one).

The idea is to make building for prpl-server easier. See Polymer/prpl-server-node#24 for motivation. Right now you need need to annotate each build in your `polymer.json` with `basePath: true` to make prpl-server work. The problem with this is that:

Reference: https://github.com/Polymer/polymer-cli/pull/920

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -165,6 +165,11 @@ export interface ProjectOptions {
    */
   builds?: ProjectBuildOptions[];
 
+  /**
+   * Tag all build with `basePath: true`.
```
Comment (aomarks) [inline src/index.ts]: How about:

  Set `basePath: true` on all builds. See that option for more details.

File: src/index.ts [hunk 2]
```diff
@@ -343,6 +349,13 @@ export class ProjectConfig {
         this.builds = this.builds.map(applyBuildPreset);
       }
     }
+
+    /**
+     * autoBasePath
+     */
+    if (options.autoBasePath) {
+      this.autoBasePath = options.autoBasePath;
```
Comment (aomarks) [inline src/index.ts]: Hmm, maybe we should just update all the builds here. I don't see why not. Then we won't need https://github.com/Polymer/polymer-build/pull/279 at all.

File: test/index_test.js [hunk 3]
```diff
@@ -593,7 +593,7 @@ suite('Project Config', () => {
   suite('json validation', () => {
     test('throws good error messages', () => {
       try {
-        ProjectConfig.validateAndCreate({ lint: [] });
+        ProjectConfig.validateAndCreate({lint: []});
```
Comment (aomarks) [inline test/index_test.js]: We don't usually format test files, can you undo this change? Maybe our `npm format` is misconfigured to include it?

Comment (abdonrd) [reply test/index_test.js]: Probably. Updated!

File: src/index.ts (hunk 2 above)
Comment (abdonrd) [reply src/index.ts]: Updated!

File: src/index.ts (hunk 1 above)
Comment (abdonrd) [reply src/index.ts]: Updated!

File: src/index.ts [hunk 4]
```diff
@@ -343,6 +349,19 @@ export class ProjectConfig {
...
+      if (Array.isArray(this.builds)) {
```
Comment (aomarks) [inline src/index.ts]: I think something like this should be fine:

```ts
for (const build of this.builds || []) {
  build.basePath = true;
}
```

The methods that read `polymer.json` run the JSON schema validator before calling this constructor, so I think we can assume types are safe.

Comment (abdonrd) [reply src/index.ts]: Ok! I had been inspired by the builds section. Updating...

**Comment by abdonrd** [comment]
@aomarks ready! 🙂

**Comment by aomarks** [comment]
Thanks for the PR!

@arthurevans @katejeffreys We should update https://www.polymer-project.org/2.0/docs/tools/polymer-json to document this new option.

**Comment by abdonrd** [comment]
@aomarks thanks! Can you release a new version? 🙂

[polymer-project-config](https://github.com/Polymer/polymer-project-config) > [polymer-build](https://github.com/Polymer/polymer-project-config) > [polymer-cli](https://github.com/Polymer/polymer-cli)

**Comment by aomarks** [comment]
Released as `3.7.0`.

**Comment by aomarks** [comment]
> @arthurevans @katejeffreys We should update https://www.polymer-project.org/2.0/docs/tools/polymer-json to document this new option.

Filed https://github.com/Polymer/docs/issues/2425 to track adding documentation.

**Comment by abdonrd** [comment]
> Released as `3.7.0`.

Thanks @aomarks!
